annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -614,8 +631,13 @@ private SolrQuery solrUniquenessQuerySimpleTagging(Annotation anno, boolean noSe
 
 	private SolrQuery solrUniquenessQueryLinkForContributing(Annotation anno, boolean noSelfDupplicate) {
       SolrQuery query = new SolrQuery();
-      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"");
-      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.LINKFORCONTRIBUTING.getOaType() + "\"");
+      query.setQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.LINKFORCONTRIBUTING.getOaType() + "\"");
+      List<String> targetValues = extractUriValues(anno.getTarget());
+      if(targetValues!=null) {
+        for(String target: targetValues) {
+          query.addFilterQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + target + "\"");
```

Comment:
this should probably swtichted to target_record_uri field, needs to be confirmed

Reply:
see comments for solrUniquenessQuerySemanticTagging

Reply:
Improved.